Model/Payment.php

```diff
@@ -277,6 +277,7 @@ public function fetchTransactionInfo(InfoInterface $payment, $transactionId)
                 $order = $payment->getOrder();
                 $this->orderHelper->updateOrderPayment($order, null, $transactionReference);
             }
+
```

Comment:
@danac-gs Is my understanding correct in that every time you fetch the transaction data from Bolt, you check whether an update is needed and update the payment according?  (I would imagine this could or does serve the role of implementing an merchant-side "Update Payment" button where the merchant syncs his actions with Bolt.)

Reply:
@mccottry Correct. In M2 there is a "Fetch" button on the transaction page that does exactly that.